app/src/main/java/org/simple/clinic/patient/SimpleVideoModule.kt

```diff
@@ -18,13 +22,24 @@ class SimpleVideoModule {
 
   @Provides
   @Named("training_video_youtube_id")
-  fun provideSimpleVideoUrlBasedOnLocale(locale: Locale): String {
-    return when (locale.language) {
-      "hi" -> "nHsQ06tiLzw"
-      "am" -> "05dnOSyHXYs"
-      "om" -> "yca12Ec5Fg4"
-      // Default to English
-      else -> "YO3D1paAuqU"
-    }
+  fun provideSimpleVideoUrlBasedOnLocale(
+      uuidSetJsonAdapter: JsonAdapter<Set<Languages>>,
+      remoteConfigService: RemoteConfigService,
+      locale: Locale
+  ): String {
+
+    val simpleVideoUrlsJsonArray: String = remoteConfigService.reader().string("simple_youtube_video_urls", "[]")
+    val simpleLanguageBasedVideoUrls: Set<Languages> = uuidSetJsonAdapter.fromJson(simpleVideoUrlsJsonArray)!!
```

Comment:
I see that the current configuration JSON for this is an array of dictionaries:

```json
[
  {
    "languageLocaleCode": "hi",
    "languageVideoUrl": "nHsQ06tiLzw"
  },
  {
    "languageLocaleCode": "am",
    "languageVideoUrl": "05dnOSyHXYs"
  },
  {
    "languageLocaleCode": "om",
    "languageVideoUrl": "yca12Ec5Fg4"
  },
  {
    "languageLocaleCode": "en",
    "languageVideoUrl": "YO3D1paAuqU"
  }
]
```

What do you think of representing this as a single dictionary instead?

```json
{
  "hi": "nHsQ06tiLzw",
  "en": "YO3D1paAuqU"
}
```

This would simplify the code required to read this a lot.

```kotlin
val stringMapAdapter = moshi.adapter<Map<String, String>>(type)
val defaultVideoJson = """
{
  "_default": "YO3D1paAuqU"
}
"""

val youtubeVideoIds: Map<String, String> = stringMapAdapter.parse(remoteConfigReader.string("simple_youtube_video_urls", defaultVideoJson))

return youtubeVideoIds[locale.language] ?: youtubeVideoIds["_default"]
```

Reply:
This would also mean that we need to add the `_default` as a key to the JSON on the remote config as well and it will handle cases where the current language is not present in the remote config as well as the app not being able to fetch the remote config,

Reply:
Oh this works as well 👍
I'll update it to the list of single dictionary

Reply:
A nitpick: can we rename the remote config key to reflect that it contains YouTube video IDs and not the URLs?

Reply:
Hey defining a type with an array looked something like  `val type = Types.newParameterizedType(Set::class.java, Languages::class.java)`
With the single dictionary, what would the raw type be? I am not able to add Map<String,String> to it.

Reply:
It's a vararg parameter, so it should be `Types.newParameterizedType(Map::class.java, String::class.java, String::class.java)`.